cli/dev/dependency.go

```diff
@@ -397,7 +402,7 @@ func fetchLocalProjectDependency(homeDirectory string, project string, userInput
 	// this project's workload definition and updates it with the reference to the ms. In the files that are read and
 	// then written we want those to preserve the env vars as env vars.
 	envVarSetting := os.Getenv("HZN_DONT_SUBST_ENV_VARS")
-	os.Setenv("HZN_DONT_SUBST_ENV_VARS", "1")
+	os.Setenv("HZN_DONT_SUBST_ENV_VARS", "0")
```

Comment:
This is because with the new changes, the service.definition.json and userinput.json contain the env variables that are defined in the hzn.cfg under the same directory.  We want the env variables to be substituted in these files for the dependent service before being pulled over to the parent. And the second  (HZN_DONT_SUBST_ENV_VARS =1) is to keep the env variables in the parent's  service.definition.json and userinput.json unchanged.   Since both parent and the children all have the env variables with the same names such as "SERVICE_NAME" "SERVICE_VERSION" etc, setting env such as SERVICE_NAME before calling this function does not work.  In the next feature, hzn will substitute the env in these files with their correct hzn.cfg file so that the user does not have to run `source` manually.

Reply:
explained offline